### PR TITLE
JBPM-5233: Data source management

### PR DESCRIPTION
**Pull request by wmedvede**
- Adds the hability of defining data base drivers that will be automatically installed during server startup.

### CODE REVIEW

**Comment by wmedvede** [comment]
@manstis Hello Mike, would you mind review this PR?

Thanks!

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/impl/DefaultDriverInitializerImpl.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        initializeFromSystemProperties();
+        initializeFromConfigFile();
+    }
+
+    protected void initializeFromSystemProperties() {
+        initializeFromProperties( System.getProperties() );
+    }
+
+    protected void initializeFromConfigFile() {
+        InputStream inputStream = DefaultDriverInitializerImpl.class.getResourceAsStream( "/datasource-management.properties" );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/impl/DefaultDriverInitializerImpl.java]: You should close your stream after use.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/impl/DefaultDriverInitializerImpl.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                driverCodes.add( propertyName.substring( UUID.length() + 1, propertyName.length() ) );
+            }
+        }
+
+        for ( String driverCode : driverCodes ) {
+            driverDef = new DriverDef();
+            if ( ( value = getDriverParam( properties, UUID, driverCode ) ) != null ) {
+                driverDef.setUuid( value );
+            } else {
+                break;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/impl/DefaultDriverInitializerImpl.java]: Do you want to ```break``` or ```continue```? Same comment for the other ```breaks``` below...

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-wildfly/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/wildfly/DeploymentIdGenerator.java [hunk 3]
```diff
@@ -54,6 +59,22 @@ public static String generateDeploymentId( final DataSourceDef dataSourceDef ) {
...
     public static String generateDeploymentId( final String uuid ) {
-        return "kie-"+ uuid;
+        return "kie" + SEPARATOR + uuid + SEPARATOR;
     }
+
+    private static final boolean isKieGenerated( String value ) {
+        return KIE_GENERATED_ID.matcher( value ).matches();
+    }
+
+    public static void main(String args[]) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-wildfly/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/core/wildfly/DeploymentIdGenerator.java]: I suspect this code was "pre-test" and can now be deleted?

**Pull request closed without merging**